akka-actor/src/main/scala/akka/actor/Address.scala

```diff
@@ -23,7 +23,7 @@ import scala.compat.java8.OptionConverters._
  * information with an address, then this must be done externally.
  */
 @SerialVersionUID(1L)
-final case class Address private (protocol: String, system: String, host: Option[String], port: Option[Int]) {
+final case class Address private[actor] (protocol: String, system: String, host: Option[String], port: Option[Int]) {
```

Comment:
The downside if using `private[actor]` here is that we don't have a good way to document that this is not intended to be part of the Java API - however the use of `Option` might be good enough of a hint here...

Reply:
Ah. I have just known that `private[akka]` will make the constructor public in java :)
scalac should make it `package private` (`default` in java) :(

Reply:
`private` also makes it public for java, so this doesn't change that aspect
```
  public akka.actor.Address(java.lang.String, java.lang.String, scala.Option<java.lang.String>, scala.Option<java.lang.Object>);
  public akka.actor.Address(java.lang.String, java.lang.String);
  public akka.actor.Address(java.lang.String, java.lang.String, java.lang.String, int);
```

Reply:
so Dotty doesn't allow private constructors at all that can be used from companion, or this is a limitation only for case classes? I assume the latter, otherwise there would be more of these changes.

Reply:
`Address`' private constructor cause `copy` and `apply` private
scala/scala#7702
lampepfl/dotty#6671